django/contrib/postgres/indexes.py

```diff
@@ -1,8 +1,40 @@
 from __future__ import unicode_literals
 
-from django.db.models import Index
+from django.db.models.indexes import Index
 
-__all__ = ['GinIndex']
+__all__ = ['BrinIndex', 'GinIndex']
+
+
+class BrinIndex(Index):
+    suffix = 'brin'
+
+    def __init__(self, fields=[], name=None, pages_per_range=None):
+        if pages_per_range is not None and pages_per_range <= 0:
+            raise TypeError('BRIN indexes require a positive integer for pages_per_range')
```

Comment:
I'd go with `ValueError` and possibly add a check `isinstance(pages_per_range, int)`:

```python
if pages_per_range is not None and not (isinstance(pages_per_range, int) and pages_per_range > 0):
    raise ValueError('pages_per_range must be None or a positive integer for BRIN indexes')
```